x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/LocalStateEnterpriseSearch.java

```diff
@@ -0,0 +1,144 @@
+/*
```

Comment:
Note: This class currently isn't used. I had tried to create an extension of the `LocalStateCompositeXPackPlugin` thinking that would be a way to return a non-null license state. I still think it's possible there's a way to do this, so I'm keeping this in for now as it's a draft PR. For now though, I'm relying on a null check in the query builder.